### ISSUE TITLE
prometheus-operator pod repeatedly writes info logs (~4 lines every second) 

### ISSUE THREAD

**Issue by nirforer** [coreos/prometheus-operator#2889]
**What did you do?**
Installed prometheus-operator on my GKE cluster

**Environment**
GKE

* Prometheus Operator version:
    `v0.29.0`

* Kubernetes version information:

Master version 1.13.11-gke.14
Node version 1.12.8-gke.6

* Kubernetes cluster kind:

   GKE

* Prometheus Operator Logs:

```
level=info ts=2019-11-28T14:28:58.487809954Z caller=operator.go:970 component=prometheusoperator msg="sync prometheus" key=monitoring/k8s
level=info ts=2019-11-28T14:28:58.86918082Z caller=operator.go:970 component=prometheusoperator msg="sync prometheus" key=monitoring/k8s
level=info ts=2019-11-28T14:28:59.397255918Z caller=operator.go:970 component=prometheusoperator msg="sync prometheus" key=monitoring/k8s
level=info ts=2019-11-28T14:29:00.491106789Z caller=operator.go:970 component=prometheusoperator msg="sync prometheus" key=monitoring/k8s
level=info ts=2019-11-28T14:29:00.874271684Z caller=operator.go:970 component=prometheusoperator msg="sync prometheus" key=monitoring/k8s
level=info ts=2019-11-28T14:29:01.402931351Z caller=operator.go:970 component=prometheusoperator msg="sync prometheus" key=monitoring/k8s
level=info ts=2019-11-28T14:29:01.682536349Z caller=operator.go:970 component=prometheusoperator msg="sync prometheus" key=monitoring/k8s
level=info ts=2019-11-28T14:29:02.497578216Z caller=operator.go:970 component=prometheusoperator msg="sync prometheus" key=monitoring/k8s
level=info ts=2019-11-28T14:29:02.880596386Z caller=operator.go:970 component=prometheusoperator msg="sync prometheus" key=monitoring/k8s
level=info ts=2019-11-28T14:29:02.926545226Z caller=operator.go:970 component=prometheusoperator msg="sync prometheus" key=monitoring/k8s
level=info ts=2019-11-28T14:29:03.409053966Z caller=operator.go:970 component=prometheusoperator msg="sync prometheus" key=monitoring/k8s
level=info ts=2019-11-28T14:29:04.503517352Z caller=operator.go:970 component=prometheusoperator msg="sync prometheus" key=monitoring/k8s
level=info ts=2019-11-28T14:29:04.886211137Z caller=operator.go:970 component=prometheusoperator msg="sync prometheus" key=monitoring/k8s
level=info ts=2019-11-28T14:29:05.42440056Z caller=operator.go:970 component=prometheusoperator msg="sync prometheus" key=monitoring/k8s
level=info ts=2019-11-28T14:29:06.508724197Z caller=operator.go:970 component=prometheusoperator msg="sync prometheus" key=monitoring/k8s
```

**Anything else we need to know?**:

The above log lines are getting written once every 1/4 of a second (~4 log lines per second) and we cannot figure out what is triggering this. I looked at the latest version of the operator.go source code and I see that there were no changes over there with regards to this log line printing.

**Comment by brancz** [comment]
Can you upgrade to latest prometheus operator? I recall us having some problem about reconciling too aggressively, which sometimes resulted in cyclic updates, where one reconcile triggered the next one.

**Comment by smueller18** [comment]
I just stumbled over this issue because my prometheus operator had the same behaviour (about 4 syncs per second). In my case, a certificate request from [cert-manager](https://github.com/jetstack/cert-manager) was in a weird state which resulted in continous putting certificate values in the etcd store. After I deleted the misbehaving `certificates.cert-manager.io` object, the prometheus operator stopped the excessive sync.

I figured it out while I was watching every change in etcd:
```
$ etcdctl version                                                                                                                                      
etcdctl version: 3.4.3
API version: 3.4
$ etcdctl watch "" --prefix=true
```
Unfortunatley, I could't reproduce a state where the prometheus operator syncs excessive again.
